api/tests/test_cloud_record.py

```diff
@@ -23,6 +23,54 @@ def setUp(self):
         """Prevent logging from appearing in test output."""
         logging.disable(logging.CRITICAL)
 
+    def test_cloud_record_post_provider_banned(self):
+        """
+        Test a banned provider cannot POST.
+
+        Even if they are on the provider list.
+        """
+        example_dn = "/C=XX/O=XX/OU=XX/L=XX/CN=allowed_host.test"
+        with self.settings(BANNED_FROM_POST=[example_dn]):
+
+            # Mock the functionality of the provider list
+            # Used in the underlying POST handling method
+            CloudRecordView._get_provider_list = Mock(return_value=PROVIDERS)
+
+            test_client = Client()
+            response = test_client.post(reverse('CloudRecordView'),
+                                        MESSAGE,
+                                        content_type="text/plain",
+                                        HTTP_EMPA_ID="Test Process",
+                                        SSL_CLIENT_S_DN=example_dn)
+
+            self.assertEqual(response.status_code, 403)
+
+    def test_cloud_record_post_provider_special(self):
+        """
+        Test a provider ganted POST rights can, indeed, POST.
+
+        Even if they aren't on the provider list.
+        This test DOES NOT check the saved message is correct.
+        For that, see test_cloud_record_post_202()
+        """
+        example_dn = "/C=XX/O=XX/OU=XX/L=XX/CN=special_host.test"
+        with self.settings(ALLOWED_FOR_POST=[example_dn],
+                           QPATH=QPATH_TEST):
+
+            # Mock the functionality of the provider list
+            # Used in the underlying POST handling method
+            CloudRecordView._get_provider_list = Mock(return_value=PROVIDERS)
+
+            test_client = Client()
+            response = test_client.post(reverse('CloudRecordView'),
+                                        MESSAGE,
+                                        content_type="text/plain",
+                                        HTTP_EMPA_ID="Test Process",
+                                        SSL_CLIENT_S_DN=example_dn)
+
+            self.assertEqual(response.status_code, 202)
+            self._delete_messages(QPATH_TEST)
```

Comment:
I think the `tearDown` doc string should be considered wrong. As not all tests in the file result in a message being saved to disk it perhaps doesn't fit in `tearDown`.

Can one create a context manager on a directory?